src/services/restaurant.services.js

```diff
@@ -1,42 +1,130 @@
+// }
+// const getRestaurantByName = async (name)=> {
+//     return await Restaurant.findOne({name}).populate('category.dishes');
+// }
+
+// const addRestaurant = async (restaurant) => {
+//     try {
+//         const newRest = await restaurant.save();   
+//     } catch (error) {
+//         console.log(error);
+//     }
+// }
+
+// const removeRestaurant = async (name)=> {
+//     await Restaurant.deleteOne({name});
+// }
+
+// const updateRestaurant = async (name, newName=name, desc, street, city, zip, phone, image) =>{
+//     const address = {street, city, zip};
+//     const openingHours = [{day: "a", hours: "1200-1600"},{day: "b", hours: "1200-1600"}]
+//     const category = [{title: "category", subtitle: "category"}]
+//     await Restaurant.findOneAndUpdate({name}, {name:newName, 
+//         description: desc, 
+//         address, 
+//         phone, 
+//         image, 
+//         category, 
+//         openingHours})
+// }
+
+// const RestaurantApi = {getAllRestaurant, getRestaurantById, getRestaurantByName, addRestaurant, removeRestaurant, updateRestaurant};
+
+
+/**
+ * 
+ * @param {*} Parameters | restaurant object
+ * @returns | return new restaurant
+ */
+const add = async (restaurant) => {
+    try{
+        const newRestaurant = await restaurant.save();
+        if(newRestaurant){
+            return newRestaurant._id;
+        }
+    }catch (error) {
+        console.log(error);
+    }
 }
 
-const getAllRestaurant = async ()=> {
-    return await Restaurant.find();
+/**
+@param {*} id type String | the id of the restaurant that we want get from database
+@returns type Restaurant | the restaurant from the database
+ */
+const getById = async (id) => {
+    try{
+        const restaurant = Restaurant.findById(id);
+        if(restaurant){
+            return restaurant;
+        }
+    }catch (error) {
+        console.log(error);
+    }
 }
-const getRestaurantByName = async (name)=> {
-    return await Restaurant.findOne({name}).populate('category.dishes');
+
+/**
+ * @param {*} Parameters type String | restaurant name
+ * @returns | return restaurant
+ */
+const getByName = async (name)=>{
+    try{
+        const restaurant = await Restaurant.findOne({name})
+        if(restaurant){
+            return restaurant;
+        }
+    }catch (error){
+        console.log(error);
+    }
 }
 
-const addRestaurant = async (restaurant) => {
-    try {
-        const newRest = await restaurant.save();   
-    } catch (error) {
+/**
+ * @returns |  return all restaurants
+ */
+const getAll = async ()=>{
+    try{
+        const restaurants = await Restaurant.find();
+        if(restaurants){
+            return restaurants;
+        }
+    }catch (error){
         console.log(error);
     }
 }
 
-const removeRestaurant = async (name)=> {
-    await Restaurant.deleteOne({name});
+/**
+ * @param {*} Parameters type String | restaurant id, restaurant object
+ * @returns | return updated restaurant
```

Comment:
change "return updated restaurant" to "type String | the id of the updated restaurant"